Checkout: Update colors, spacing, and typography for new design

sirbrillig opened the pull request:
## Proposed Changes

This PR modifies the appearance of checkout in relatively small ways (although the visual effect is considerable) to match an updated design. This should not really change any functionality; checkout's step behavior should remain the same before and after this change.

This is part of a larger project of redesigning checkout: p9Jlb4-6FB-p2 Specifically this is the first milestone: pbOQVh-3pj-p2

## Screenshots

### Before

<img width="1280" alt="Screenshot 2023-06-20 at 6 49 42 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/6de9716f-b583-41c1-8761-c199f1f922bd">

<img width="1275" alt="jetpack-before" src="https://github.com/Automattic/wp-calypso/assets/2036909/605b9e52-2786-4378-99fc-a11ca9916d34">

### After

<img width="1279" alt="Screenshot 2023-06-20 at 6 26 35 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/1a4c0435-6a39-4c4a-8219-daab01e5b4fd">

<img width="1273" alt="jetpack-after" src="https://github.com/Automattic/wp-calypso/assets/2036909/b47aef9e-64be-4f0a-a45c-284e49f061b7">


## Testing Instructions

This is a hard thing to test completely because checkout has so many variations. This should not change any functionality, so the first thing is to make sure that checkout renders successfully. Beyond that, it's a good idea to test any variation you can think of. Here are some suggestions:

- Checkout (and any of the other variations) at wide ("desktop") widths.
- Checkout (and any of the other variations) at around 900px wide ("tablet") widths.
- Checkout (and any of the other variations) at around 500px wide ("mobile") widths.
- Checkout for a Jetpack product.
- Checkout for an Akismet product.
- Checkout for a Gift product.
- Checkout with and without having a coupon added.
- Checkout with and without a domain in the cart.
- Checkout with an empty cart.
- Checkout after completing a synchronous (ie: non-3DS credit card) payment before being redirected.
- Checkout with a third-party-developer (eg: Yoast) plugin in the cart.
- Checkout in an RTL locale.
- Checkout with a monthly, yearly, and biennial WPCom plan in the cart.

Some non-checkout things that should be tested as well because they may be affected by this:

- Adding a payment method to an existing subscription.
- Viewing the mini-cart in the masterbar.
- Viewing the one-click checkout upsell after checkout completes.

sirbrillig committed:
Add className to Checkout Contact Form step

sirbrillig commented:
Question that I think is not covered by the Figma designs: what should the hover color for the buttons be?

sirbrillig commented:
Before/After currently:

Before

<img width="644" alt="Screenshot 2023-05-31 at 6 43 32 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/2ec95062-4861-4c57-87e5-f2f802fa1941">

After

<img width="669" alt="Screenshot 2023-05-31 at 6 43 10 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/b8f71195-cb3f-4894-9767-d58890052569">

sirbrillig commented:
To do:

- Bold product names
- Header "Checkout"
- "Included in purchase" format total/tax same as main column
- left align "By checking out" with "i" icons.
- Maybe? font-weight to 500 instead of 600 for bold
- Masterbar color etc.
- Maybe? button radius to 4
- checkmarks need to be adjusted with size and color
- when contact step is active, fix appearance of payment step to be greyed out
- fix mobile view

michaeldcain commented:
There were some layout styles leaking in from the Reader, preventing the page from being full-width. I opened https://github.com/Automattic/wp-calypso/pull/78128 to fix the issue.

sirbrillig commented:
Current UI:

<img width="1180" alt="Screenshot 2023-06-13 at 5 59 07 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/140894cd-8c70-4473-981b-c414aa3bed32">

sirbrillig commented:
As of 852d84f:


| Before | After |
|--------|--------|
| <img width="1157" alt="Screenshot 2023-06-14 at 5 48 23 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/ca6ff5e4-f70c-4df2-86b9-2f31ea40cbff"> | <img width="1288" alt="Screenshot 2023-06-15 at 7 55 46 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/d7735c50-b444-48bc-b830-b21d04a88c21"> |
| <img width="622" alt="Screenshot 2023-06-14 at 5 51 37 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/2adf128f-6a3a-42ed-b849-ff36963ace15"> | ![calypso localhost_3000_checkout_paytontest171 wordpress com(iPad Air)](https://github.com/Automattic/wp-calypso/assets/2036909/b5fe2b8a-dd11-45d2-97b9-606153e47d4d) |
| ![wordpress com_checkout_paytontest171 wordpress com_bypassTargetRedirection=true(iPhone XR)](https://github.com/Automattic/wp-calypso/assets/2036909/e6fd1853-41ce-4407-b9ad-fbc1f3310a76) | ![calypso localhost_3000_checkout_paytontest171 wordpress com(iPhone XR)](https://github.com/Automattic/wp-calypso/assets/2036909/f468140f-73d4-445b-a080-a87e02851585) |

sirbrillig commented on packages/composite-checkout/src/components/button.tsx:
```diff
@@ -7,7 +7,7 @@ const CallToAction = styled( 'button' )< CallToActionProps >`
 	width: ${ ( props: CallToActionProps ) => ( props.fullWidth ? '100%' : 'auto' ) };
 	font-size: 16px;
-	border-radius: ${ ( props ) => ( props.buttonType === 'paypal' ? '50px' : '2px' ) };
+	border-radius: 4px;
```
We should double-check the PayPal UI after this change.

sirbrillig replied:
Added to the checklist to verify.

michaeldcain replied:
Yeah, I kind of liked the consistency of the more square button. Since we're updating the logo, we're still within [PayPal's button guidelines](https://developer.paypal.com/docs/log-in-with-paypal/customize/button-design-guide/). We don't really need to do the color either, but the yellow is pretty identifiable.

sirbrillig commented:
Current view:

<img width="1279" alt="Screenshot 2023-06-20 at 6 26 35 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/b97beb0d-a53e-402c-867a-3fd25f1d8d12">

<img width="617" alt="Screenshot 2023-06-20 at 6 27 41 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/5240aa2c-5f75-462d-a6dd-f9728de4bb5c">

<img width="416" alt="Screenshot 2023-06-20 at 6 27 25 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/14f25009-92ca-4b4e-b015-d752d9a85143">

sirbrillig committed:
Add className to CheckoutNextStepButton

sirbrillig committed:
Add color-studio to composite-checkout deps

sirbrillig committed:
Restyle checkout masterbar

sirbrillig committed:
Restyle payment methods

sirbrillig committed:
Restyle checkout

sirbrillig committed:
Change tests to use "Continue to payment" button

sirbrillig commented:
Rebased and squashed most of the commits together (the latter was necessary due to some pretty potent conflicts which git rebase could not even understand).

sirbrillig commented:
As a side-effect, this changes the "Change Payment Method" radio buttons. Hopefully that's ok.

## Before
<img width="961" alt="Screenshot 2023-06-20 at 7 07 02 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/c87a77e7-b1e4-472b-aae5-aebe25891d66">

## After
<img width="982" alt="Screenshot 2023-06-20 at 7 06 49 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/b53aefaa-f417-4832-babb-e0279aa207a1">

sirbrillig committed:
Undo accidental test changes for modal Continue buttons

sirbrillig committed:
Hide TaxLabel for PayPal if empty

sirbrillig committed:
Update credit-card test to account for "now" in Pay button

sirbrillig committed:
Move "Billing information" to the same line on change payment method

sirbrillig commented:
Updated the "Change Payment Method" form to move "Billing information" to the same line as the card info for saved credit cards at wide widths:

<img width="724" alt="Screenshot 2023-06-21 at 10 31 36 AM" src="https://github.com/Automattic/wp-calypso/assets/2036909/ba72327a-aaec-4512-bf3b-4c594b7aaf26">

michaeldcain reviewed (approved):
This works well for me at different screen resolutions from mobile to desktop, and with different product combinations and services.

I tested:
- siteless Jetpack checkout for a few different product combinations
- Jetpack checkout with the pre-purchase notice showing
- siteless Akismet checkout
- WPcom checkout with plans, bundled domains, multiple emails, Marketplace products

spraveenitpro commented:
<details>
<summary> Checkout (and any of the other variations) at wide ("desktop") widths: ✅ </summary>

 ![e6Gezt.png](https://github.com/Automattic/wp-calypso/assets/552016/0f69650c-05e9-4186-b216-189deac1f74d) LGTM 
</details>


<details>
<summary> Checkout (and any of the other variations) at around 900px wide ("tablet") widths:  IPAD Air 768 X 1024: ✅ </summary>

![R7jCgI.png](https://github.com/Automattic/wp-calypso/assets/552016/270f2643-207c-41a8-91d5-68f23b87948b)✅ 
</details>

<details>
<summary> IPAD mini [768X1024]: shows additional white space in the header : 🎈</summary>

![YERGpA.png](https://github.com/Automattic/wp-calypso/assets/552016/a91b6a35-57cf-4f4a-9bca-31264ce84f39)✅ 
</details>


<details>
<summary>Checkout for a Jetpack product.✅  </summary>

![FhfLLT.png](https://github.com/Automattic/wp-calypso/assets/552016/537f41ae-def6-4223-8fb2-2715206b9847)
</details>

<details>
<summary>Checkout for an Akismet product. ✅ </summary>

 ![55Fr5T.png](https://github.com/Automattic/wp-calypso/assets/552016/6b424543-b84b-43ee-ab31-416ffe92c8e7)

</details>

<details>
<summary>Checkout with and without having a coupon added. ✅ </summary>

![orVxtF.png](https://github.com/Automattic/wp-calypso/assets/552016/50b13e6d-2374-4a5c-8e71-4e362015d535)
</details>

<details>
<summary>Checkout with and without a domain in the cart. ✅ </summary>


![eXERzb.png](https://github.com/Automattic/wp-calypso/assets/552016/6b5da8bf-2b46-42a0-954c-6846775c39ff)
</details>

<details>
<summary> Checkout with an empty cart. ✅ </summary>
![czRwl7.png](https://github.com/Automattic/wp-calypso/assets/552016/1dcdfaf8-51c6-4812-8215-966d3c867782)
</details>

<details>
<summary>Checkout with a third-party-developer (eg: Yoast) plugin in the cart. ✅ </summary>

![dLetWU.png](https://github.com/Automattic/wp-calypso/assets/552016/6baad11a-f69d-4081-91d4-17d2396326e8)
</details>

<details>
<summary>Checkout in an RTL locale. ✅ </summary>

![ylPOsw.png](https://github.com/Automattic/wp-calypso/assets/552016/3c22c561-e35b-4c54-b6aa-0e4025bbc460)
</details>

<details>
<summary>Checkout with a monthly, yearly, and biennial WPCom plan in the cart. ✅ </summary>

![9gaKWv.png](https://github.com/Automattic/wp-calypso/assets/552016/51b4c3ef-aea8-49d0-aec7-cf906a8a2b89)

![1DjFU1.png](https://github.com/Automattic/wp-calypso/assets/552016/7209a973-e350-4f60-8940-8d291256646f)

![k4xePN.png](https://github.com/Automattic/wp-calypso/assets/552016/37f775e8-1bc6-409e-9801-fcb0bc81077e)
</details>

<details>
<summary>Checkout after completing a synchronous (ie: non-3DS credit card) payment before being redirected. ✅ </summary>

![JGlAND.gif](https://github.com/Automattic/wp-calypso/assets/552016/f22e22f9-8f29-4e2e-9f84-9ec9c8f0ebed)
</details>

spraveenitpro reviewed (approved):
I did some static analysis, and nothing significant stood out. Very well done 👌🏾

sirbrillig committed:
Increase specificity of removal of layout padding for checkout

sirbrillig commented:
> IPAD mini [768X1024]: shows additional white space in the header 

Thanks for spotting that. It's being caused by a breakpoint's layout padding which is overriding the `padding: 0` for checkout. 

<img width="492" alt="Screenshot 2023-06-21 at 5 33 05 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/2fa924cd-ecd1-4f69-9e3f-3104f76e7493">

It's a little tricky to solve because I can only reproduce it on the calypso.live links, not in development. I assume it has to do with CSS load order or something? In any case, I tried to increase the specificity of the `padding: 0`in e5dc93e so we'll see if that fixes it.

sirbrillig committed:
Do not render CheckoutHelpLink container if it should be hidden

sirbrillig commented:
Rebased again to include the removal of HappyChat from #78433 since that was causing an unfortunate whitespace to appear with this branch.

<img width="682" alt="Screenshot 2023-06-21 at 5 51 46 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/0c64405b-e739-46d7-99c7-9077a3f9f215">

I believe I've resolved that space in afeb15f

michaeldcain committed:
Fix Checkout padding override

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8062237

Thank you @sirbrillig for including a screenshot in the description! This is really helpful for our translators.

sirbrillig commented:
One bug reported and hopefully fixed: https://github.com/Automattic/wp-calypso/pull/78544

sirbrillig commented:
Another regression: https://github.com/Automattic/wp-calypso/pull/78592

sirbrillig commented:
A style race condition: https://github.com/Automattic/wp-calypso/pull/78687

sirbrillig commented:
Another style race condition: https://github.com/Automattic/wp-calypso/pull/78760

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8062237).
